pennylane/data/data_manager/__init__.py

```diff
@@ -56,17 +56,43 @@ def _get_data_struct():
 
 
 def _download_partial(
-    s3_url: str, dest: Path, attributes: typing.Iterable[str], overwrite: bool
+    s3_url: str, dest: Path, attributes: Optional[typing.Iterable[str]], overwrite: bool
 ) -> None:
```

Comment:
in the scenario when we're downloading a partial but all keys are already present, can we consult the local dataset and return early? I'm not _totally_ against `attributes` being optional, but it feels weird given what this function does. we should not have to go to the database at all to compare attributes if the dataset is present and has what we need.

Reply:
okay, updated `_download_partial` to do this and added a test for this behaviour